sync_diff_inspector/diff.go

```diff
@@ -302,19 +302,24 @@ func (df *Diff) checkChunkDataEqual(checkJobs []*CheckJob, table *TableCheckCfg)
 }
 
 func (df *Diff) compareRows(rows1, rows2 *sql.Rows, orderKeyCols []*model.ColumnInfo, table *TableCheckCfg) (bool, error) {
-	equal := true
-	rowsData1 := make([]map[string][]byte, 0, 100)
-	rowsData2 := make([]map[string][]byte, 0, 100)
+	var (
+		equal        = true
+		null1, null2 map[string]bool
+		data1, data2 map[string][]byte
+		rowsData1    = make([]map[string][]byte, 0, 100)
+		rowsData2    = make([]map[string][]byte, 0, 100)
+		err          error
+	)
 
 	for rows1.Next() {
-		data1, err := dbutil.ScanRow(rows1)
+		data1, null1, err = dbutil.ScanRow(rows1)
```

Comment:
Don’t need to update null1? Only update rowsData1?

Reply:
my mistake, already done